cache_ns_zone.go

```diff
@@ -126,6 +137,31 @@ func (zone *NSCacheZone) Index() {
 	zone.s = nsCacheGlueMap(zone.glue)
 }
 
+// ReplyNS produces a response message equivalent to
+// an NS request for the cache domain, including the
+// known glue and current TTL.
+func (zone *NSCacheZone) ReplyNS(req *dns.Msg) *dns.Msg {
+	ttl := zone.TTL()
+
+	zone.mu.Lock()
+	defer zone.mu.Unlock()
+
+	resp := new(dns.Msg)
+	resp.SetReply(req)
```

Comment:
Here you set the `resp` to be the response for `req` hence `Question` section of `resp` is `[]dns.Question{req.Question[0]}` then below you change that to a `NS` request. If `req.QType != dns.TyxpeNS` you are asking for trouble. But all tests pass for now, probably we will experience issues when the function is called effectivelly.

Reply:
@karasz This method isn't part of the standard operations of NSCache and I was thinking of it just as a "standard-ish" way to export the data in the `NSCacheZone`. I'll introduce a similar ReplyNS() on NSCache itself and remove the overriding of the question in the next PR